src/__tests__/integration/basic/pst-kv.test.ts

```diff
@@ -0,0 +1,122 @@
+import fs from 'fs';
+
+import ArLocal from 'arlocal';
+import Arweave from 'arweave';
+import { JWKInterface } from 'arweave/node/lib/wallet';
+import path from 'path';
+import { mineBlock } from '../_helpers';
+import { PstState, PstContract } from '../../../contract/PstContract';
+import { InteractionResult } from '../../../core/modules/impl/HandlerExecutorFactory';
+import { Warp } from '../../../core/Warp';
+import { WarpFactory } from '../../../core/WarpFactory';
+import { LoggerFactory } from '../../../logging/LoggerFactory';
+
+describe('Testing the Profit Sharing Token', () => {
+  let contractSrc: string;
+
+  let wallet: JWKInterface;
+  let walletAddress: string;
+
+  let initialState: PstState;
+
+  let arweave: Arweave;
+  let arlocal: ArLocal;
+  let warp: Warp;
+  let pst: PstContract;
+  let pstVM: PstContract;
+
+  beforeAll(async () => {
+    // note: each tests suit (i.e. file with tests that Jest is running concurrently
+    // with another files has to have ArLocal set to a different port!)
+    arlocal = new ArLocal(2222, false);
+    await arlocal.start();
+    LoggerFactory.INST.logLevel('error');
+
+    warp = WarpFactory.forLocal(2222);
+
+    ({ arweave } = warp);
+    ({ jwk: wallet, address: walletAddress } = await warp.generateWallet());
+
+    contractSrc = fs.readFileSync(path.join(__dirname, '../data/kv-storage.js'), 'utf8');
+    const stateFromFile: PstState = JSON.parse(fs.readFileSync(path.join(__dirname, '../data/token-pst.json'), 'utf8'));
+
+    initialState = {
+      ...stateFromFile,
+      ...{
+        owner: walletAddress
+      }
+    };
+
+    // deploying contract using the new SDK.
+    const { contractTxId } = await warp.deploy({
+      wallet,
+      initState: JSON.stringify(initialState),
+      src: contractSrc
+    });
+
+    // connecting to the PST contract
+    pst = warp.pst(contractTxId).setEvaluationOptions({
+      useKVStorage: true
+    }) as PstContract;
+    pst.connect(wallet);
+
+    await mineBlock(warp);
+  });
+
+  afterAll(async () => {
+    await arlocal.stop();
+    // TODO: remove the kv storage folders
+  });
+
+  it('should initialize', async () => {
```

Comment:
we need to decide how the kv storage should be initialized...those mints are just to set the values, that the standard pst.test.ts sets via initial state.